src/Http/Offer/CurrentUserHasPermissionRequestHandler.php

```diff
@@ -33,6 +34,6 @@ public function handle(ServerRequestInterface $request): ResponseInterface
 
         $hasPermission = $this->hasPermission($offerId, $this->currentUserId);
 
-        return new JsonResponse($hasPermission, StatusCodeInterface::STATUS_OK, $this->getPrivateHeaders());
+        return new UncacheableJsonResponse($hasPermission, StatusCodeInterface::STATUS_OK);
```

Comment:
Couple of concerns with this class:

- `Uncacheable` is incorrect, because you set `Cache-Control: private` which means that it should be cached but only in private caches, for example browser caches. It must not be stored in shared caches, for example the caches of proxies.
- This response is for a request that uses an `Authorization` header. But responses to requests with an Authorization header cannot be stored in shared caches by design, since they are unique for a given user. So the `Cache-Control: private` is not necessary / useful I think.

(See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)

Reply:
The name has been updated to `PrivateJsonResponse` to better describe to actual function.
I've kept it the same as the old code where the header is also set. Since the uses `Authorization` it might indeed be a bit too secure or unnecessary

Reply:
See https://github.com/cultuurnet/udb3-silex/pull/1116#discussion_r959298493